app/src/main/java/com/dessalines/thumbkey/utils/Utils.kt

```diff
@@ -241,13 +243,17 @@ fun getImeActionCode(ime: IMEService): Int {
 /**
  * Returns the correct keyboard mode
  */
-fun getKeyboardMode(ime: IMEService): KeyboardMode {
+fun getKeyboardMode(ime: IMEService): (AppSettings?) -> KeyboardMode {
```

Comment:
I don't know how to work properly with `@Composable`, so I couldn't extract the app settings in `IMEService` to pass `autoCapitalize` it here. So I made it to return a lambda.

Reply:
This  is a function. You pass in parameters by using function parameters. So provide it with either the settings, or settings?.autocapitalize.